### ISSUE TITLE
[Feature] Add support for moving torrents to a different directory when completed

### ISSUE THREAD

**Issue by Chaz6** [varbhat/exatorrent#2]
It would be useful to be able to have torrents moved to a directory when they have finished downloading. For example, when added, torrents are saved to the directory "incomplete" and when finished they are moved to the directory "complete". It would probably also be prudent to check if a torrent exists in the complete folder when adding in case it has already been downloaded.

**Comment by onedr0p** [comment]
I don't understand why close this issue? It's a valid feature request. Closing it means you A) Have implemented or fixed the issue or B) Will not implement or fix the issue.

**Comment by varbhat** [comment]
@onedr0p ,

Sorry. This feature will soon be implemented soon (it will be another daemon program which you run along with exatorrent).

I wanted to ask whether symlinking works good(i am thinking of implementing 'move' too)

**Comment by onedr0p** [comment]
What limitations are there to implementing it directly into exatorrent? It seems weird to run a separate service to handle this one task.

**Comment by varbhat** [comment]
@onedr0p , Feature can be implemented , but i want it to be implemented "correctly" .

exatorrent is:
1) multi user program.
2) uses `<exadir>/torrents/<infohash>`  to save torrents. This pattern is used by exatorrent to implement many features.
3) feature must be carefully thought out and must not interfere with above ones.

So, thinking of webhooks/hooks which will be called when torrent completes. So, instead of integrating move feature into exatorrent itself,let itself be small other program.

**Comment by onedr0p** [comment]
I'm really confused by the multi-user feature of exatorrent and it honestly complicates too much for a torrent client. Containers are by far the most popular way to deploy applications and adoption is growing everyday. It's very easy to spin up a container for each user with external auth  (like authentik / Authellia) instead of trying to make it multi-tennant. For me I love simplicity and less code. However, this is more of an architectural discussion.

**Comment by varbhat** [comment]
>> multi-user feature ... complicates too much
Nope, it won't complicate much. Also,this feature is needed for some of users of exatorrent (including me)

>> I love simplicity and less code
You are free to read the source code and decide yourself

>> this is more of an architectural discussion
Yeah. I would like exatorrent to be simple , powerful and also extensible.  There are many things that can be done on event completion. So, instead of stuffing all those into exatorrent, i would suggest a API and program to deal with it however you need.

Also, exatorrent operates within a directory and at no time, it will operate out of this directory. It's better if this rule is adhered to.

I will be implementing this feature soon.
